### ISSUE TITLE
FailedAttachVolume

### ISSUE THREAD

**Issue by danielstankw** [KusionStack/karpor#550]
### What happened?

There is an issue with the helm chart deployment and Volume.

When deploying the helm chart the following happens:

```bashrc
elasticsearch-5654f4f8c5-8xjkj   1/1     Running             0              16m
etcd-0                           0/1     ContainerCreating   0              13m
karpor-server-d9d477955-9gb4t    0/1     CrashLoopBackOff    7 (3m ago)     16m
karpor-syncer-64d449f4c-fbzjv    0/1     Error               8 (5m7s ago)   16m
```

This is due:
`k describe pod etcd-0`

```bashrc
  Warning  FailedAttachVolume  12m   attachdetach-controller  AttachVolume.Attach failed for volume "pvc-f28692d1-1d3e-4b8a-b2fc-1980dc2c1eee" : InvalidVolume.NotFound: The volume 'vol-03e1c30abe533984b' does not exist.
           status code: 400, request id: ef0543fb-03dd-4a9c-82ab-06f2cca0faa2
```

`k get pvc`
```bashrc
NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-etcd-0   Bound    pvc-f28692d1-1d3e-4b8a-b2fc-1980dc2c1eee   10Gi       RWO            gp2            3h59m

```

### What did you expect to happen?

....

### How can we reproduce it (as minimally and precisely as possible)?

Run the default helm chart with unchanged values.

### Karpor version

<details>

```console
$ karpor -V
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

**Comment by elliotxx** [comment]
Oh, I'm not in front of the computer right now. I'll reproduce it locally later. 

Does this mistake exist all the time, or will it just wait a while?

**Comment by danielstankw** [comment]
@elliotxx 
For me its been there all the time. 
It seems like its trying to use volume that doesnt exist.

**Comment by elliotxx** [comment]
Strange, we didn't encounter this case during the test. Let me see what's going on.

**Comment by danielstankw** [comment]
@elliotxx any update on this?

**Comment by ruquanzhao** [comment]
Hi, @danielstankw, welcome to the Community!

It seems you are using a k8s cluster with AWS. Can you please check your `Persistent Volume`? If you don't have an available `Persistent Volume`, you might need to either configure a `StorageClass` or manually create a 10GB storage volume.

**Comment by danielstankw** [comment]
@ruquanzhao thanks for reply,
It seems like issue is happening due to the difference in AWS setup.
Would it be possible for you to add fields for defining pvc in the helm chart? - that would solve the issue

**Comment by ruquanzhao** [comment]
> @ruquanzhao thanks for reply, It seems like issue is happening due to the difference in AWS setup. Would it be possible for you to add fields for defining pvc in the helm chart? - that would solve the issue

Yes, you are right, the stroageclass can be configured through Helm chart. Let's discuss more for details!
WDYT, @elliotxx ?

**Comment by elliotxx** [comment]
@ruquanzhao @danielstankw Yeah, PVC has been mentioned multiple times (#347 #522 ...), and we will support specifying it in helm this week!

**Comment by rajeswarr** [comment]
> > @ruquanzhao thanks for reply, It seems like issue is happening due to the difference in AWS setup. Would it be possible for you to add fields for defining pvc in the helm chart? - that would solve the issue
> 
> Yes, you are right, the stroageclass can be configured through Helm chart. Let's discuss more for details! WDYT, @elliotxx ?

is this implemented, still I am getting the issue?

**Comment by elliotxx** [comment]
> > > @ruquanzhao thanks for reply, It seems like issue is happening due to the difference in AWS setup. Would it be possible for you to add fields for defining pvc in the helm chart? - that would solve the issue
> > 
> > 
> > Yes, you are right, the stroageclass can be configured through Helm chart. Let's discuss more for details! WDYT, @elliotxx ?
> 
> is this implemented, still I am getting the issue?

@rajeswarr Thank you feedback. Have you encountered this problem too? We will provide the ability to customize StorageClass this week.

**Comment by rajeswarr** [comment]
> > > > @ruquanzhao thanks for reply, It seems like issue is happening due to the difference in AWS setup. Would it be possible for you to add fields for defining pvc in the helm chart? - that would solve the issue
> > > 
> > > 
> > > Yes, you are right, the stroageclass can be configured through Helm chart. Let's discuss more for details! WDYT, @elliotxx ?
> > 
> > 
> > is this implemented, still I am getting the issue?
> 
> @rajeswarr Thank you feedback. Have you encountered this problem too? We will provide the ability to customize StorageClass this week.

@elliotxx , I got this issue on-premises cluster.

**Comment by danielstankw** [comment]
@elliotxx if you want I can modify the helm chart and create pull request, unless you started on this topic?

**Comment by elliotxx** [comment]
> @elliotxx if you want I can modify the helm chart and create pull request, unless you started on this topic?

@danielstankw  Very welcome! As I replied in another issue, I can only start optimizing the chart a few days later. If you can start directly, that would be great.

**Comment by danielstankw** [comment]
@elliotxx sounds good, will do

**Comment by rajeswarr** [comment]
> @elliotxx sounds good, will do

Thank you @elliotxx  and @danielstankw. Please let me know once changes are done.

**Comment by elliotxx** [comment]
> > @elliotxx sounds good, will do
> 
> Thank you @elliotxx and @danielstankw. Please let me know once changes are done.

@rajeswarr Of course 👌

**Comment by wenxuanwu** [comment]
I also encountered a similar problem and can temporarily solve it by creating a local pv

**Comment by elliotxx** [comment]
> I also encountered a similar problem and can temporarily solve it by creating a local pv

Yeah, We are following this problem
